### ISSUE TITLE
Allow different arrow-body-style setting for callbacks

### ISSUE THREAD

**Issue by Lioness100** [eslint/eslint#16501]
### What rule do you want to change?

arrow-body-style

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
[1, 2, 3].map((num) => num * 2);

const multiplyByThree = (arr) => {
  return arr.map((num) => num * 2);
};
```

### What does the rule currently do for this code?

Currently, eslint would either:

1. `"off"` - Ignore both
2. `"always"` - Transform the first statement to use curly braces
3. `"never"` or `"as-needed"` - Transform the second statement to not use curly braces

### What will the rule do after it's changed?

I'm looking for an option that allows me to set different behavior depending on whether the function is a callback. In my case, for example, `as-needed` for callbacks (because `[1, 2, 3].map((num) => { return num * 2; });` seems lengthy), and `always` for functions
 I assign to a variable (just because that's how I like my arrow functions ;). Especially if I have multiple multiline functions in a row, it looks weird if I have a function in the middle without braces)

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @Lioness100, thanks for the issue! 

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we are not adding new options to the `arrow-body-style` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.
